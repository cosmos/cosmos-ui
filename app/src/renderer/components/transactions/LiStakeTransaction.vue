<template>
  <li-transaction
    color="#47AB6C"
    :time="transaction.time"
    :block="transaction.height"
  >
    <template v-if="txType === `cosmos-sdk/MsgCreateValidator`">
      <div slot="caption">
        Create validator&nbsp;
        <b>{{ full(atoms(tx.value.amount)) }}</b>
        <span>&nbsp;{{ tx.value.denom }}s</span>
      </div>
      <div slot="details">
        Moniker:&nbsp;
        <router-link :to="`${url}/${tx.validator_address}`">
          {{
            moniker(tx.validator_address)
          }}
        </router-link>
      </div>
    </template>
    <template v-else-if="txType === `cosmos-sdk/MsgEditValidator`">
      <div slot="caption">
        Edit validator&nbsp;
        <router-link :to="`${url}/${tx.validator_address}`">
          {{
            moniker(tx.validator_address)
          }}
        </router-link>
      </div>
    </template>
    <template v-else-if="txType === `cosmos-sdk/MsgDelegate`">
      <div slot="caption">
        Delegation&nbsp;
        <b>{{ full(atoms(tx.value.amount)) }}</b>
        <span>&nbsp;{{ tx.value.denom }}s</span>
      </div>
      <div slot="details">
        To&nbsp;
        <router-link :to="`${url}/${tx.validator_address}`">
          {{
            moniker(tx.validator_address)
          }}
        </router-link>
      </div>
    </template>
    <template v-else-if="txType === `cosmos-sdk/MsgBeginRedelegate`">
      <div slot="caption">
        Redelegation&nbsp;
        <b>
          {{
            calculatePrettifiedTokens(tx.validator_src_address, tx.shares_amount)
          }}
        </b>
        <span>&nbsp;{{ bondingDenom }}s</span>
      </div>
      <div slot="details">
        From&nbsp;
        <router-link :to="`${url}/${tx.validator_src_address}`">
          {{
            moniker(tx.validator_src_address)
          }}
        </router-link>
        &nbsp;to&nbsp;
        <router-link :to="`${url}/${tx.validator_dst_address}`">
          {{
            moniker(tx.validator_dst_address)
          }}
        </router-link>
      </div>
    </template>
    <template v-else-if="txType === `cosmos-sdk/MsgUndelegate`">
      <div slot="caption">
        Undelegation&nbsp;
        <b>
          {{
            calculatePrettifiedTokens(tx.validator_address, tx.shares_amount)
          }}
        </b>
        <span>&nbsp;{{ bondingDenom }}s</span>
        <template v-if="timeDiff">
          <span class="tx-unbonding__time-diff">
            &nbsp;{{ timeDiff }}
          </span>
        </template>
      </div>
      <div slot="details">
        From&nbsp;
        <router-link :to="`${url}/${tx.validator_address}`">
          {{
            moniker(tx.validator_address)
          }}
        </router-link>
      </div>
    </template>
    <template v-else-if="txType === `cosmos-sdk/MsgUnjail`">
      <div slot="caption">
        Unjail&nbsp;
        <router-link :to="`${url}/${tx.address}`">
          {{
            moniker(tx.address)
          }}
        </router-link>
      </div>
    </template>
  </li-transaction>
</template>

<script>
import LiTransaction from "./LiTransaction"
import { atoms, full } from "../../scripts/num.js"
import { calculateTokens } from "../../scripts/common.js"
import moment from "moment"

/*
 * undelegation tx need a preprocessing, where shares are translated into transaction.balance: {amount, denom}
 */

export default {
  name: `li-stake-transaction`,
  components: { LiTransaction },
  props: {
    transaction: {
      type: Object,
      required: true
    },
    validators: {
      type: Array,
      required: true
    },
    url: {
      type: String,
      required: true
    },
    bondingDenom: {
      type: String,
      required: true
    },
    unbondingTime: {
      type: Number,
      default: null
    },
    txType: {
      type: String,
      required: true
    }
  },
  data: () => ({
    atoms,
    full
  }),
  computed: {
    tx() {
      return this.transaction.tx.value.msg[0].value
    },
    timeDiff() {
      // only show time diff if still waiting to be terminated
      if (this.state !== `locked`) return ``

      return `(liquid ${moment(this.unbondingTime).fromNow()})`
    },
    // unbonding transactions can be in the state 'locked', 'ended'
    // the transaction needs to be enriched from the outside with `unbondingDelegation`
    state() {
      if (!this.unbondingTime) return `ended`
      if (this.unbondingTime - Date.now() <= 0) return `ended`
      return `locked`
    }
  },
  methods: {
    moniker(validatorAddr) {
      const validator = this.validators.find(
        c => c.operator_address === validatorAddr
      )
      return validator ? validator.description.moniker : validatorAddr
    },
    calculatePrettifiedTokens(validatorAddr, shares) {
      const validator = this.validators.find(
        val => val.operator_address === validatorAddr
      )
      return this.atoms(calculateTokens(validator, shares).toNumber())
    }
  }
}
</script>
