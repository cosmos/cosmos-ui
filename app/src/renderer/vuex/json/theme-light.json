{
  "app-fg": "#eee",
  "app-bg": "#fff",
  "app-bg-alpha": "hsla(233, 0%, 0%, 5%)",
  "bright": "#000",
  "txt": "#333",
  "dim": "#666",
  "bc": "#ddd",
  "bc-dim": "#eee",
  "hover-bg": "hsl(233, 0%, 90%)",
  "input-bc": "hsl(233, 22%, 67%)",
  "input-bc-hover": "hsl(233, 22%, 40%)",
  "primary-bc": "#99f",
  "primary": "val(--app-bg)"
}
