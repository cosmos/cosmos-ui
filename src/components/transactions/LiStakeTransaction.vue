<template>
  <LiTransaction
    color="#47AB6C"
    :time="time"
    :block="block"
    :memo="memo"
    :fees="fees"
  >
    <template v-if="txType === `cosmos-sdk/MsgCreateValidator`">
      <div slot="caption">
        Create validator
        <b>{{ tx.amount.amount | toAtoms }}</b>
        <span>{{ value | viewDenom }}</span>
      </div>
      <div slot="details">
        Moniker:
        <router-link :to="`${url}/${tx.validator_address}`">
          {{ moniker(tx.validator_address) }}
        </router-link>
      </div>
    </template>
    <template v-else-if="txType === `cosmos-sdk/MsgEditValidator`">
      <div slot="caption">
        Edit validator
      </div>
      <div slot="details">
        Moniker:
        <router-link :to="`${url}/${tx.validator_address}`">
          {{ moniker(tx.validator_address) }}
        </router-link>
      </div>
    </template>
    <template v-else-if="txType === `cosmos-sdk/MsgDelegate`">
      <div slot="caption">
        Delegated
        <b>{{ tx.amount.amount | toAtoms }}</b>
        <span>{{ tx.amount.denom | viewDenom }}</span>
      </div>
      <div slot="details">
        To&nbsp;
        <router-link :to="`${url}/${tx.validator_address}`">
          {{ moniker(tx.validator_address) }}
        </router-link>
      </div>
    </template>
    <template v-else-if="txType === `cosmos-sdk/MsgUndelegate`">
      <div slot="caption">
        Undelegated
        <b>
          {{ tx.amount.amount | toAtoms }}
        </b>
        <span>{{ bondingDenom | viewDenom }}</span>
        <template v-if="timeDiff">
          <span class="tx-unbonding__time-diff">
            {{ timeDiff }}
          </span>
        </template>
      </div>
      <div slot="details">
        From&nbsp;
        <router-link :to="`${url}/${tx.validator_address}`">
          {{ moniker(tx.validator_address) }}
        </router-link>
      </div>
    </template>
    <template v-else-if="txType === `cosmos-sdk/MsgBeginRedelegate`">
      <div slot="caption">
        Redelegated
        <b>
          {{ tx.amount.amount | toAtoms }}
        </b>
        <span>{{ bondingDenom | viewDenom }}</span>
      </div>
      <div slot="details">
        From&nbsp;
        <router-link :to="`${url}/${tx.validator_src_address}`">
          {{ moniker(tx.validator_src_address) }}
        </router-link>
        &nbsp;to&nbsp;
        <router-link :to="`${url}/${tx.validator_dst_address}`">
          {{ moniker(tx.validator_dst_address) }}
        </router-link>
      </div>
    </template>
    <template v-else-if="txType === `cosmos-sdk/MsgUnjail`">
      <div slot="caption">
        Unjail
      </div>
      <div slot="details">
        Moniker:
        <router-link :to="`${url}/${tx.address}`">
          {{ moniker(tx.address) }}
        </router-link>
      </div>
    </template>
  </LiTransaction>
</template>

<script>
import LiTransaction from "./LiTransaction"
import { atoms as toAtoms, viewDenom } from "../../scripts/num.js"
import moment from "moment"
import { formatBech32 } from "src/filters"

/*
 * undelegation tx need a preprocessing, where shares are translated into transaction.balance: {amount, denom}
 */

export default {
  name: `li-stake-transaction`,
  components: { LiTransaction },
  filters: {
    toAtoms,
    viewDenom
  },
  props: {
    tx: {
      type: Object,
      required: true
    },
    fees: {
      type: Object,
      required: true
    },
    validators: {
      type: Array,
      required: true
    },
    url: {
      type: String,
      required: true
    },
    bondingDenom: {
      type: String,
      required: true
    },
    unbondingTime: {
      type: Number,
      default: null
    },
    txType: {
      type: String,
      required: true
    },
    time: {
      type: String,
      required: true
    },
    block: {
      type: Number,
      required: true
    },
    memo: {
      type: String,
      default: null
    }
  },
  computed: {
    timeDiff() {
      // only show time diff if still waiting to be terminated
      if (this.state !== `locked`) return ``

      return `(liquid ${moment(this.unbondingTime).fromNow()})`
    },
    // unbonding transactions can be in the state 'locked', 'ended'
    // the transaction needs to be enriched from the outside with `unbondingDelegation`
    state() {
      if (!this.unbondingTime) return `ended`
      if (this.unbondingTime - Date.now() <= 0) return `ended`
      return `locked`
    },
    value() {
      return (this.tx.value && this.tx.value.denom) || ""
    }
  },
  methods: {
    moniker(validatorAddr) {
      const validator = this.validators.find(
        c => c.operator_address === validatorAddr
      )
      return validator
        ? validator.description.moniker
        : formatBech32(validatorAddr)
    }
  }
}
</script>
