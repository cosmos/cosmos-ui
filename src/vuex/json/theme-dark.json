{
  "app-fg": "hsl(233, 36%, 18%)",
  "app-bg": "hsl(233, 30%, 22%)",
  "app-nav": "hsl(233, 38%, 14%)",
  "tertiary": "hsl(325, 96%, 59%)",
  "secondary": "hsl(277, 92%, 58%)",
  "primary": "hsl(233, 88%, 57%)",
  "link": "hsl(217, 100%, 70%)",
  "bright": "hsl(0, 100%, 100%)",
  "hover": "hsla(0, 0%, 0%, 0)",
  "txt": "hsla(0, 100%, 100%, 0.8)",
  "dim": "hsla(0, 100%, 100%, 0.667)",
  "bc": "hsla(233, 24%, 75%, 0.175)",
  "bc-dim": "hsla(233, 24%, 75%, 0.0875)"
}
