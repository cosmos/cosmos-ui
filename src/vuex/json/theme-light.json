{
  "app-fg": "hsl(0,0%,92%)",
  "app-bg": "#f5f5f5",
  "app-nav": "#ddd",
  "tertiary": "hsl(325, 96%, 59%)",
  "secondary": "hsl(277, 92%, 58%)",
  "primary": "hsl(233, 88%, 57%)",
  "link": "hsl(233, 88%, 57%)",
  "bright": "hsl(0, 100%, 100%)",
  "hover": "hsla(0, 0%, 0%, 0)",
  "bright": "#000",
  "txt": "hsl(233, 38%, 14%)",
  "dim": "#3b3b3b",
  "bc": "#ddd",
  "bc-dim": "#cdcbcb"
}
