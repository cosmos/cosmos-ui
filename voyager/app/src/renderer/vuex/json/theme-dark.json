{
  "app-fg": "hsl(233, 33%, 16%)",
  "app-bg": "hsl(233, 36%, 13%)",
  "app-bg-alpha": "hsla(233, 36%, 13%, 95%)",
  "bright": "#fff",
  "txt": "hsl(233, 13%, 85%)",
  "dim": "hsl(233, 13%, 60%)",
  "bc": "hsl(233, 22%, 23%)",
  "bc-dim": "hsl(233, 33%, 16%)",
  "hover-bg": "hsl(233, 43%, 10%)",
  "input-bc": "hsl(233, 22%, 33%)",
  "input-bc-hover": "hsl(233, 22%, 40%)"
}
