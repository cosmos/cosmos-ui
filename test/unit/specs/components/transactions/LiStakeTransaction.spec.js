import { shallowMount } from "@vue/test-utils"
import LiStakeTransaction from "transactions/LiStakeTransaction"
import { stakingTxs } from "../../store/json/txs"
import { state } from "renderer/connectors/lcdClientMock.js"

describe(`LiStakeTransaction`, () => {
  let wrapper
  const validators = state.candidates
  const propsData = {
    tx: stakingTxs[0].tx.value.msg[0].value,
    txType: `cosmos-sdk/MsgCreateValidator`,
    validators,
    url: `/validator`,
    bondingDenom: `uatom`,
    fees: {
      amount: `3421`,
      denom: `uatom`
    },
    time: new Date(Date.now()).toISOString(),
    block: 500
  }

  beforeEach(() => {
    wrapper = shallowMount(LiStakeTransaction, { propsData, stubs: [`router-link`] })
  })

  describe(`create validator`, () => {

    it(`with fees`, () => {
      expect(wrapper.vm.$el).toMatchSnapshot()
    })

    it(`without fees`, () => {
      wrapper.setProps({
        fees: null
      })
      expect(wrapper.vm.$el).toMatchSnapshot()
    })
  })

  describe(`edit validator`, () => {

    beforeEach(() => {
      wrapper.setProps({
        tx: stakingTxs[1].tx.value.msg[0].value,
        txType: `cosmos-sdk/MsgEditValidator`,
        fees: {
          amount: `3421`,
          denom: `uatom`
        }
      })
    })

    it(`with fees`, async () => {
      expect(wrapper.vm.$el).toMatchSnapshot()
    })

    it(`without fees`, () => {
      wrapper.setProps({
        fees: null
      })
      expect(wrapper.vm.$el).toMatchSnapshot()
    })
  })

  describe(`delegations`, () => {
    beforeEach(() => {
      wrapper.setProps({
        tx: stakingTxs[2].tx.value.msg[0].value,
        txType: `cosmos-sdk/MsgDelegate`,
        fees: {
          amount: `3421`,
          denom: `uatom`
        }
      })
    })

    it(`with fees`, () => {
      expect(wrapper.vm.$el).toMatchSnapshot()
    })

    it(`without fees`, () => {
      wrapper.setProps({
        fees: null
      })
      expect(wrapper.vm.$el).toMatchSnapshot()
    })

    it(`should show the validator address if no validator with that address in store`, () => {
      wrapper.setProps({
        validators: []
      })
      expect(
        wrapper
          .text()
          .indexOf(
            `To cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw`
          ) === -1
      ).toBe(true)
    })
  })

  describe(`unbonding delegations`, () => {
    beforeEach(() => {
      wrapper.setProps({
        tx: stakingTxs[3].tx.value.msg[0].value,
        txType: `cosmos-sdk/MsgUndelegate`,
        fees: {
          amount: `3421`,
          denom: `uatom`
        }
      })
    })

    it(`with fees`, () => {
      wrapper.setProps({
        unbondingTime: Date.now() + 1000
      })

      expect(wrapper.contains(`.tx-unbonding__time-diff`)).toBe(true)
      expect(wrapper.vm.$el).toMatchSnapshot()
    })

    it(`without fees`, () => {
      wrapper.setProps({
        fees: null
      })
      expect(wrapper.vm.$el).toMatchSnapshot()
    })

    it(`should show unbonding delegations as ended`, () => {
      wrapper.setProps({ unbondingTime: Date.now() - 1000 })
      expect(wrapper.vm.$el).toMatchSnapshot()
      expect(wrapper.text()).toContain(`0.000323`)
    })

    it(`should default to ended if no unbonding delegation is present`, () => {
      expect(wrapper.vm.$el).toMatchSnapshot()
    })
  })

  describe(`redelegations`, () => {

    beforeEach(() => {
      wrapper.setProps({
        tx: stakingTxs[4].tx.value.msg[0].value,
        txType: `cosmos-sdk/MsgBeginRedelegate`,
        fees: {
          amount: `3421`,
          denom: `uatom`
        }
      })
    })

    it(`with fees`, () => {
      expect(wrapper.vm.$el).toMatchSnapshot()
      expect(wrapper.text()).toContain(`30.000000`)
    })

    it(`without fees`, () => {
      wrapper.setProps({
        fees: null
      })

      expect(wrapper.vm.$el).toMatchSnapshot()
    })
  })

  describe(`unjail`, () => {
    beforeEach(() => {
      wrapper.setProps({
        tx: stakingTxs[5].tx.value.msg[0].value,
        txType: `cosmos-sdk/MsgUnjail`,
        fees: {
          amount: `3421`,
          denom: `uatom`
        }
      })
    })

    it(`with fees`, () => {
      expect(wrapper.vm.$el).toMatchSnapshot()
    })

    it(`without fees`, () => {
      wrapper.setProps({
        fees: null
      })
      expect(wrapper.vm.$el).toMatchSnapshot()
    })
  })
})
