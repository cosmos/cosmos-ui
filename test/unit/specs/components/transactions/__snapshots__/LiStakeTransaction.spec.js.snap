// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiStakeTransaction create validator with fees 1`] = `
<li-transaction-stub
  block="193281"
  color="#47AB6C"
  time="1970-01-01T00:00:02.042Z"
>
  <div>
    
      Create validator
      
    <b>
      1.000000
    </b>
     
    <span>
      uatoms
    </span>
  </div>
   
  <div>
    
      Moniker:
      
    <router-link-stub
      to="/validator/cosmosvaloper1qecshyc40kshszkwrtscgmsdd8tz3n4hrj9yf2"
    >
      
        cosmosvaloper1qecshyc40kshszkwrtscgmsdd8tz3n4hrj9yf2
      
    </router-link-stub>
  </div>
   
  <div>
    
      Network Fee: 
    <b>
      0.003421
    </b>
     
    <span>
      uatoms
    </span>
  </div>
</li-transaction-stub>
`;

exports[`LiStakeTransaction create validator without fees 1`] = `
<li-transaction-stub
  block="193281"
  color="#47AB6C"
  time="1970-01-01T00:00:02.042Z"
>
  <div>
    
      Create validator
      
    <b>
      1.000000
    </b>
     
    <span>
      uatoms
    </span>
  </div>
   
  <div>
    
      Moniker:
      
    <router-link-stub
      to="/validator/cosmosvaloper1qecshyc40kshszkwrtscgmsdd8tz3n4hrj9yf2"
    >
      
        cosmosvaloper1qecshyc40kshszkwrtscgmsdd8tz3n4hrj9yf2
      
    </router-link-stub>
  </div>
   
  <div>
    
      Network Fee: 
    <b>
      0.000000
    </b>
     
    <span>
      uatoms
    </span>
  </div>
</li-transaction-stub>
`;

exports[`LiStakeTransaction delegations with fees 1`] = `
<li-transaction-stub
  block="568"
  color="#47AB6C"
  time="1970-01-01T00:00:42.000Z"
>
  <div
    slot="caption"
  >
    
      Delegation
      
    <b>
      42,000.000000
    </b>
     
    <span>
      uatoms
    </span>
  </div>
   
  <div
    slot="details"
  >
    
      To
      
    <router-link-stub
      to="/validator/cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw"
    >
      
        mr_mounty
      
    </router-link-stub>
  </div>
   
  <div
    slot="fees"
  >
    
      Network Fee: 
    <b>
      0.003421
    </b>
     
    <span>
      uatoms
    </span>
  </div>
</li-transaction-stub>
`;

exports[`LiStakeTransaction delegations without fees 1`] = `
<li-transaction-stub
  block="568"
  color="#47AB6C"
  time="1970-01-01T00:00:42.000Z"
>
  <div
    slot="caption"
  >
    
      Delegation
      
    <b>
      42,000.000000
    </b>
     
    <span>
      uatoms
    </span>
  </div>
   
  <div
    slot="details"
  >
    
      To
      
    <router-link-stub
      to="/validator/cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw"
    >
      
        mr_mounty
      
    </router-link-stub>
  </div>
   
  <div
    slot="fees"
  >
    
      Network Fee: 
    <b>
      0.000000
    </b>
     
    <span>
      uatoms
    </span>
  </div>
</li-transaction-stub>
`;

exports[`LiStakeTransaction edit validator with fees 1`] = `
<li-transaction-stub
  block="193281"
  color="#47AB6C"
  time="1970-01-01T00:00:02.041Z"
>
  <div
    slot="caption"
  >
    
      Edit validator
    
  </div>
   
  <div
    slot="details"
  >
    
      Moniker:
      
    <router-link-stub
      to="/validator/cosmosvaloper1qecshyc40kshszkwrtscgmsdd8tz3n4hrj9yf2"
    >
      
        cosmosvaloper1qecshyc40kshszkwrtscgmsdd8tz3n4hrj9yf2
      
    </router-link-stub>
  </div>
   
  <div
    slot="fees"
  >
    
      Network Fee: 
    <b>
      0.003421
    </b>
     
    <span>
      uatoms
    </span>
  </div>
</li-transaction-stub>
`;

exports[`LiStakeTransaction edit validator without fees 1`] = `
<li-transaction-stub
  block="193281"
  color="#47AB6C"
  time="1970-01-01T00:00:02.041Z"
>
  <div
    slot="caption"
  >
    
      Edit validator
    
  </div>
   
  <div
    slot="details"
  >
    
      Moniker:
      
    <router-link-stub
      to="/validator/cosmosvaloper1qecshyc40kshszkwrtscgmsdd8tz3n4hrj9yf2"
    >
      
        cosmosvaloper1qecshyc40kshszkwrtscgmsdd8tz3n4hrj9yf2
      
    </router-link-stub>
  </div>
   
  <div
    slot="fees"
  >
    
      Network Fee: 
    <b>
      0.000000
    </b>
     
    <span>
      uatoms
    </span>
  </div>
</li-transaction-stub>
`;

exports[`LiStakeTransaction redelegations with fees 1`] = `
<li-transaction-stub
  block="567"
  color="#47AB6C"
  time="1970-01-01T00:00:45.000Z"
>
  <div
    slot="caption"
  >
    
      Redelegation
      
    <b>
      
        30.000000
      
    </b>
     
    <span>
      uatoms
    </span>
  </div>
   
  <div
    slot="details"
  >
    
      From
      
    <router-link-stub
      to="/validator/cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw"
    >
      
        mr_mounty
      
    </router-link-stub>
    
      to
      
    <router-link-stub
      to="/validator/cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctplpn3au"
    >
      
        good_greg
      
    </router-link-stub>
  </div>
   
  <div
    slot="fees"
  >
    
      Network Fee: 
    <b>
      0.003421
    </b>
     
    <span>
      uatoms
    </span>
  </div>
</li-transaction-stub>
`;

exports[`LiStakeTransaction redelegations without fees 1`] = `
<li-transaction-stub
  block="567"
  color="#47AB6C"
  time="1970-01-01T00:00:45.000Z"
>
  <div
    slot="caption"
  >
    
      Redelegation
      
    <b>
      
        30.000000
      
    </b>
     
    <span>
      uatoms
    </span>
  </div>
   
  <div
    slot="details"
  >
    
      From
      
    <router-link-stub
      to="/validator/cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw"
    >
      
        mr_mounty
      
    </router-link-stub>
    
      to
      
    <router-link-stub
      to="/validator/cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctplpn3au"
    >
      
        good_greg
      
    </router-link-stub>
  </div>
   
  <div
    slot="fees"
  >
    
      Network Fee: 
    <b>
      0.000000
    </b>
     
    <span>
      uatoms
    </span>
  </div>
</li-transaction-stub>
`;

exports[`LiStakeTransaction unbonding delegations should default to ended if no unbonding delegation is present 1`] = `
<li-transaction-stub
  block="569"
  color="#47AB6C"
  time="1970-01-01T00:00:04.200Z"
>
  <div
    slot="caption"
  >
    
      Undelegation
      
    <b>
      
        0.000323
      
    </b>
     
    <span>
      uatoms
    </span>
     
    <!---->
  </div>
   
  <div
    slot="details"
  >
    
      From
      
    <router-link-stub
      to="/validator/cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw"
    >
      
        mr_mounty
      
    </router-link-stub>
  </div>
   
  <div
    slot="fees"
  >
    
      Network Fee: 
    <b>
      0.003421
    </b>
     
    <span>
      uatoms
    </span>
  </div>
</li-transaction-stub>
`;

exports[`LiStakeTransaction unbonding delegations should show unbonding delegations as ended 1`] = `
<li-transaction-stub
  block="569"
  color="#47AB6C"
  time="1970-01-01T00:00:04.200Z"
>
  <div
    slot="caption"
  >
    
      Undelegation
      
    <b>
      
        0.000323
      
    </b>
     
    <span>
      uatoms
    </span>
     
    <!---->
  </div>
   
  <div
    slot="details"
  >
    
      From
      
    <router-link-stub
      to="/validator/cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw"
    >
      
        mr_mounty
      
    </router-link-stub>
  </div>
   
  <div
    slot="fees"
  >
    
      Network Fee: 
    <b>
      0.003421
    </b>
     
    <span>
      uatoms
    </span>
  </div>
</li-transaction-stub>
`;

exports[`LiStakeTransaction unbonding delegations with fees 1`] = `
<li-transaction-stub
  block="569"
  color="#47AB6C"
  time="1970-01-01T00:00:04.200Z"
>
  <div
    slot="caption"
  >
    
      Undelegation
      
    <b>
      
        0.000323
      
    </b>
     
    <span>
      uatoms
    </span>
     
    <span
      class="tx-unbonding__time-diff"
    >
      
          (liquid in a few seconds)
        
    </span>
  </div>
   
  <div
    slot="details"
  >
    
      From
      
    <router-link-stub
      to="/validator/cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw"
    >
      
        mr_mounty
      
    </router-link-stub>
  </div>
   
  <div
    slot="fees"
  >
    
      Network Fee: 
    <b>
      0.003421
    </b>
     
    <span>
      uatoms
    </span>
  </div>
</li-transaction-stub>
`;

exports[`LiStakeTransaction unbonding delegations without fees 1`] = `
<li-transaction-stub
  block="569"
  color="#47AB6C"
  time="1970-01-01T00:00:04.200Z"
>
  <div
    slot="caption"
  >
    
      Undelegation
      
    <b>
      
        0.000323
      
    </b>
     
    <span>
      uatoms
    </span>
     
    <!---->
  </div>
   
  <div
    slot="details"
  >
    
      From
      
    <router-link-stub
      to="/validator/cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw"
    >
      
        mr_mounty
      
    </router-link-stub>
  </div>
   
  <div
    slot="fees"
  >
    
      Network Fee: 
    <b>
      0.000000
    </b>
     
    <span>
      uatoms
    </span>
  </div>
</li-transaction-stub>
`;

exports[`LiStakeTransaction unjail with fees 1`] = `
<li-transaction-stub
  block="193281"
  color="#47AB6C"
  time="1970-01-01T00:00:20.420Z"
>
  <div
    slot="caption"
  >
    
      Unjail
    
  </div>
   
  <div
    slot="details"
  >
    
      Moniker:
      
    <router-link-stub
      to="/validator/cosmosvaloper1qecshyc40kshszkwrtscgmsdd8tz3n4hrj9yf2"
    >
      
        cosmosvaloper1qecshyc40kshszkwrtscgmsdd8tz3n4hrj9yf2
      
    </router-link-stub>
  </div>
   
  <div
    slot="fees"
  >
    
      Network Fee: 
    <b>
      0.003421
    </b>
     
    <span>
      uatoms
    </span>
  </div>
</li-transaction-stub>
`;

exports[`LiStakeTransaction unjail without fees 1`] = `
<li-transaction-stub
  block="193281"
  color="#47AB6C"
  time="1970-01-01T00:00:20.420Z"
>
  <div
    slot="caption"
  >
    
      Unjail
    
  </div>
   
  <div
    slot="details"
  >
    
      Moniker:
      
    <router-link-stub
      to="/validator/cosmosvaloper1qecshyc40kshszkwrtscgmsdd8tz3n4hrj9yf2"
    >
      
        cosmosvaloper1qecshyc40kshszkwrtscgmsdd8tz3n4hrj9yf2
      
    </router-link-stub>
  </div>
   
  <div
    slot="fees"
  >
    
      Network Fee: 
    <b>
      0.000000
    </b>
     
    <span>
      uatoms
    </span>
  </div>
</li-transaction-stub>
`;
